app/post.go

```diff
@@ -231,12 +221,11 @@ func (a *App) CreatePost(post *model.Post, channel *model.Channel, triggerWebhoo
 
 	// Verify the parent/child relationships are correct
 	var parentPostList *model.PostList
-	if pchan != nil {
-		result := <-pchan
-		if result.NErr != nil {
+	if post.RootId != "" {
```

Comment:
Hmm .. creating posts are a hot path and should be fast. I _think_ I am okay with it, but it would be good to do a quick micro-benchmark to ensure we aren't losing out on performance.

Reply:
Will defo do a benchmark for these, good point.